### PR TITLE
java.time.* custom AMQP serializers and tests.  Excludes pure enums.

### PR DESCRIPTION
**Pull request by rick-r3**

### CODE REVIEW

**Comment by rick-r3** [comment]
Just realised I forgot to register them in the standard place so they are available outside the tests.  Will update that class too.

**Comment by rick-r3** [comment]
And they're all stateless so will make them all `object`.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/DurationSerializer.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import net.corda.nodeapi.internal.serialization.amqp.CustomSerializer
+import net.corda.nodeapi.internal.serialization.amqp.SerializerFactory
+import java.time.Duration
+
+/**
+ * A serializer for [Duration] that uses a proxy object to write out the seconds and the nanos.
+ */
+class DurationSerializer(factory: SerializerFactory) : CustomSerializer.Proxy<Duration, DurationSerializer.DurationProxy>(Duration::class.java, DurationProxy::class.java, factory) {
+    override val additionalSerializers: Iterable<CustomSerializer<out Any>> = emptyList()
```
Comment (vkolomeyko) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/DurationSerializer.kt]: I wonder if we should make `emptyList()` default implementation in `net.corda.nodeapi.internal.serialization.amqp.CustomSerializer.Proxy`.
There will be too many occurrences when no extra serializers will be needed.

Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/DurationSerializer.kt]: Yeah.  Sounds good.  Will do that now.

**Pull request merged**